scripts/maint/release.py

```diff
@@ -1,42 +1,171 @@
-    print("# Create the tarball signature:")
-    print("gpg --default-key F305447AF806D46B --detach-sign v{}.tar.gz  "
-          "--output v{}.tar.gz.asc")
-    print("# Upload the signature manually to Github.")
-    print("# Now bump to the next development version:")
-    print("# Replace the version in __init__.py with the next version -dev0")
-    print("# Commit the development version:")
-    print("git commit -a -m 'Bump to version X'")
-    print("# Done!")
+    print(__doc__)
+    print("1. Which version to release")
+    print("---------------------------")
+    current_version = semantic_version.Version(sbws.__version__)
+    release_version = obtain_release_version(current_version)
+
+    print("\n2. Update changelog")
+    print("-------------------")
+    print("Creating tag v{} so that gitchangelog will create the new section "
+          "with the correct new tag...".format(release_version))
+    subprocess.call(['git', 'tag', 'v{}'.format(release_version),
+                     '-m', '\"Release version {}.\"'.format(release_version)])
+
+    print("\nUpdating the changelog automatically...")
+    subprocess.call('gitchangelog'.split(' '))
+
+    print("\nEdit the changelog manually to remove merge commits and "
+          "leave only the apropiate paragraph for every bug/feature.")
+    input("Press enter when done.\n")
+
+    print("\nRemoving the tag...")
+    subprocess.call(['git', 'tag', '-d', 'v{}'.format(release_version)])
+
+    print("\n3. Modify program version")
+    print("--------------------------")
+    print("\nReplacing __init__.py version with the release version...")
+    replace_version(current_version, release_version)
+
+    print("\nCommiting __init__.py and CHANGELOG.rst...")
+    subprocess.call(['git', 'commit',
+                     '-am', '"Release version {}."'.format(release_version)])
+
+    print("\n4. Create tag and sign it")
+    print("--------------------------")
+    print("\nCreating the final tag and signing it...")
+    subprocess.call(['git', 'tag', '-s', 'v{}'.format(release_version),
+                     '-m', '"Release version {}."'.format(release_version)])
+
+    print("\n5. Push commit and tag")
+    print("----------------------")
+    print("\nPush now so that the Github tarball will have the"
+          " correct tag...")
+    input("Press enter when you are sure everything is correct.")
+    subprocess.call(['git', 'push', 'origin', 'master'])
+
+    # While we use Github releases and not dist.tpo
+    print("\n6. Create release tarball")
+    print("-------------------------")
+    print("Obtaining Github tarball...")
+    subprocess.call(
+        "torsocks wget https://github.com/torproject/sbws/archive/v{}.tar.gz"
+        .format(release_version).split(' ')
+        )
+
+    print("\n7. Create the tarball signature")
+    print("-------------------------------")
+    print("Creating detached signature...")
+    subprocess.call("gpg --default-key F305447AF806D46B "
```

Comment:
Whose key is this?
It should probably be an argument to the script.

Reply:
that's juga's key. Agree, just didn't do it cause forgot.